library/src/main/java/com/novoda/downloadmanager/LiteDownloadService.java

```diff
@@ -51,13 +51,14 @@ public IBinder onBind(Intent intent) {
     }
 
     @Override
-    public void download(final DownloadBatch downloadBatch, final DownloadBatchStatusCallback callback) {
+    public void download(DownloadBatch downloadBatch, DownloadBatchStatusCallback callback) {
         callback.onUpdate(downloadBatch.status());
 
         downloadBatch.setCallback(callback);
 
         executor.execute(() -> {
             acquireCpuWakeLock();
+            downloadBatch.persist();
```

Comment:
this is the fix, we guarantee that when the download starts, the necessary information in the database exists